### DISCUSSION TITLE
Requests being served from the accelerate cache when not explicitly set to do so.

### DISCUSSION THREAD

**Discussion by adeniyii** [prisma/prisma#23482] [Q&A]
### Question

My team has a NextJs application using `supabase` as the database provider and wired up to `prisma-accelerate` mainly for connection pooling.  We setup caching on two specific routes (one for user-info and one for workspace-info). We do this using the `withAccelerate` extension linked to a separate Prisma client, which we use selectively on those specific routes.

![CleanShot 2024-03-14 at 12  28 38@2x](https://github.com/prisma/prisma/assets/52568775/07b15392-18a1-400c-b7b3-d23e5538c876)

![CleanShot 2024-03-14 at 12  29 58@2x](https://github.com/prisma/prisma/assets/52568775/3cedbc68-5c05-49cb-bd03-d6742106493c)

We've noticed though, that another route was making use of the cache which resulted in perpetually stale data being fetched for that route. This is weird because we did not setup this route to use the accelerate-enabled Prisma client.

Even weirder is that we could not get the route to stop hitting the accelerate cache by removing all instances of the accelerate-enabled Prisma client.

We've been able to solve this in 2 ways: 
- switch out the database_url for a non-accelerate url — using the supabase url directly
- use the accelerate-enabled Prisma client on the affected route, but set the `ttl` to 0.

Option 1 is a no-op because we very much would like to keep using prisma-accelerate for it's connection pooling features.
Option 2 works, but is far from perfect. Opting into caching for a route we don't want cached in order to prevent unwanted caching (using the `ttl` option) is weird.

I'm thinking we're doing something wrong, but would appreciate some guidance in diagnosing this problem.

### Expected behavior (optional)

- Prevent using the cache for unwanted routes.
- Use the cache for the specified routes.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider  = "postgresql"
  url       = env("DATABASE_URL")
  directUrl = env("DIRECT_DATABASE_URL")
}

```

- Database: PostgreSQL
- Node.js version: 20.0.0

- Run `prisma -v` to see your Prisma version and paste it

prisma                  : 5.10.2
@prisma/client          : 5.10.2
Computed binaryTarget   : darwin-arm64
Operating System        : darwin
Architecture            : arm64
Node.js                 : v20.11.0
Query Engine (Node-API) : libquery-engine 5a9203d0590c951969e85a7d07215503f4672eb9 (at ../../node_modules/.pnpm/@prisma+engines@5.10.2/node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli 5a9203d0590c951969e85a7d07215503f4672eb9 (at ../../node_modules/.pnpm/@prisma+engines@5.10.2/node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.10.0-34.5a9203d0590c951969e85a7d07215503f4672eb9
Default Engines Hash    : 5a9203d0590c951969e85a7d07215503f4672eb9
Studio                  : 0.499.0

**Comment by ludralph** [comment]
Hi @Adeniyii 👋 

Thank you for raising this question.

This is already being worked on in our [discord servers ](https://discord.com/channels/937751382725886062/1217589758566465547/1217589758566465547). 

I'll go ahead and close this discussion.
